tuned/plugins/plugin_net.py

```diff
@@ -283,6 +291,64 @@ def _get_nf_conntrack_hashsize(self):
 			return int(value)
 		return None
 
+	def _get_re_ip_link_show(self):
+		if self._re_ip_link_show is None:
+			self._re_ip_link_show = re.compile(RE_IP_LINK_SHOW, re.MULTILINE)
+		return self._re_ip_link_show
+
+	def _call_ip_link(self, args=[]):
+		if not self._use_ip:
+			return None
+		args = ["ip", "link"] + args
+		(rc, out, err_msg) = self._cmd.execute(args, no_errors=[errno.ENOENT], return_err=True)
+		if rc == -errno.ENOENT:
+			log.warn("ip command not found, ignoring for other devices")
+			self._use_ip = False
+			return None
+		elif rc:
+			log.info("Problem calling ip command")
+			log.debug("(rc: %s, msg: '%s')" % (rc, err_msg))
+			return None
+		return out
+
+	def _ip_link_show(self, device=None):
+		"""
+		Returns dict with device/interface name as key and dict of params defined in RE_IP_LINK_SHOW as value
+		"""
+		args = ["show"]
+		if device:
+			args.append(device)
+		out = self._call_ip_link(args)
+		if out is None:
+			return None
+		result = {}
+		for x in self._get_re_ip_link_show().finditer(out):
```

Comment:
We called `ip link show $(device)`, do we need to iterate through the result?

Reply:
It's if we'd want call _ip_link_show for more then one device and for other info in the future, but it's probably adding unnecessary complexity from my side (same goes for comment above)